ext/app/electron/DocRegistry.ts

```diff
@@ -0,0 +1,74 @@
+import * as path from "path";
+import { HomeDBManager } from "app/gen-server/lib/homedb/HomeDBManager";
+import { fileExists } from "./utils";
+
+export class DocRegistry {
+
+  private idToPathMap: Map<string, string>;
+  private pathToIdMap: Map<string, string>;
+  private db: HomeDBManager;
+
+  private constructor(dbManager: HomeDBManager) {
+    this.db = dbManager;
+  }
+
+  public static async create(dbManager: HomeDBManager) {
+    // Allocate space.
+    const dr = new DocRegistry(dbManager);
+    dr.idToPathMap = new Map<string, string>;
+    dr.pathToIdMap = new Map<string, string>;
```

Comment:
The constructor was made private because you are supposed to use `create()` to create a DocRegistry. You cannot `new DocRegistry(...)` - tsc will report an error. Basically `create()` _is_ the constructor.

No strong preference here, but I think there won't be much difference.